src/API/Site/Controllers/Google/AccountController.php

```diff
@@ -83,17 +85,38 @@ public function register_routes(): void {
 	 * @return callable
 	 */
 	protected function get_connect_callback(): callable {
-		return function() {
+		return function( Request $request ) {
 			try {
+				$next = $request->get_param( 'next' );
+				$path = $next === 'setup-mc' ? '/google/setup-mc' : '/google/settings&subpath=/reconnect-accounts';
+
 				return [
-					'url' => $this->connection->connect( admin_url( 'admin.php?page=wc-admin&path=/google/setup-mc' ) ),
+					'url' => $this->connection->connect( admin_url( "admin.php?page=wc-admin&path=$path" ) ),
```

Comment:
Small nitpick, which isn't a problem since it's at the end of the string. I think it's better to wrap the variables in curly brackets.

```suggestion
					'url' => $this->connection->connect( admin_url( "admin.php?page=wc-admin&path={$path}" ) ),
```

Reply:
Thanks for the suggestion.